internal/resolve/catalog.go

```diff
@@ -190,6 +178,42 @@ func (r *CatalogResolver) Resolve(ctx context.Context, ext *ocv1alpha1.ClusterEx
 	return resolvedBundle, resolvedBundleVersion, priorDeprecation, nil
 }
 
+type resolutionErrorInfo struct {
+	PackageName     string
+	Version         string
+	Channels        []string
+	InstalledBundle *ocv1alpha1.BundleMetadata
+	ResolvedBundles []foundBundle
+}
+
+func buildResolutionError(rei resolutionErrorInfo) error {
+	var errStringBuilder strings.Builder
```

Comment:
I might be in the minority, but I'd suggest a much shorter name for this variable since:
1. It is used in like 50% of the lines in this function.
2. The whole purpose of the function is to build up a string, so a short variable name will not really lose meaning
3. IMO, it would make the code look cleaner.

```suggestion
	var sb strings.Builder
```